core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -422,6 +425,13 @@ class BlockManagerMasterEndpoint(
     if (blockLocations.containsKey(blockId)) blockLocations.get(blockId).toSeq else Seq.empty
   }
 
+  private def getLocationsAndStatus(
+      blockId: BlockId): (Seq[BlockManagerId], Option[BlockStatus]) = {
```

Comment:
shall we create a class for location and status? e.g.
```
case class BlockLocationsAndStatus(locations: Seq[BlockManagerId], status: BlockStatus) {
  assert(locations.nonEmpty)
}
```
Then this method can return `Option[BlockLocationsAndStatus]`